src/peft/peft_model.py

```diff
@@ -136,32 +143,49 @@ def save_pretrained(self, save_directory: str, safe_serialization: bool = False,
-            output_state_dict = get_peft_model_state_dict(
-                self, state_dict=kwargs.get("state_dict", None), adapter_name=adapter_name
-            )
-            output_dir = os.path.join(save_directory, adapter_name) if adapter_name != "default" else save_directory
-            os.makedirs(output_dir, exist_ok=True)
+        if selected_adapters is None:
+            selected_adapters = list(self.peft_config.keys())
+        elif selected_adapters is not None:
+            if any(
+                selected_adapter_name not in list(self.peft_config.keys())
+                for selected_adapter_name in selected_adapters
+            ):
+                raise ValueError(
+                    f"You passed an invalid `selected_adapters` arguments, current supported adapter names are"
+                    f" {list(self.peft_config.keys())} - got {selected_adapters}."
+                )
 
-            if safe_serialization:
-                safe_save_file(
-                    output_state_dict, os.path.join(output_dir, SAFETENSORS_WEIGHTS_NAME), metadata={"format": "pt"}
+        for adapter_name, peft_config in self.peft_config.items():
+            if adapter_name in selected_adapters:
+                # save only the trainable weights
+                output_state_dict = get_peft_model_state_dict(
+                    self, state_dict=kwargs.get("state_dict", None), adapter_name=adapter_name
                 )
-            else:
-                torch.save(output_state_dict, os.path.join(output_dir, WEIGHTS_NAME))
-
-            # save the config and change the inference mode to `True`
-            if peft_config.base_model_name_or_path is None:
-                peft_config.base_model_name_or_path = (
-                    self.base_model.__dict__.get("name_or_path", None)
-                    if isinstance(peft_config, PromptLearningConfig)
-                    else self.base_model.model.__dict__.get("name_or_path", None)
+                output_dir = (
+                    os.path.join(save_directory, adapter_name) if adapter_name != "default" else save_directory
                 )
-            inference_mode = peft_config.inference_mode
-            peft_config.inference_mode = True
-            peft_config.save_pretrained(output_dir)
-            peft_config.inference_mode = inference_mode
+                os.makedirs(output_dir, exist_ok=True)
+
+                if safe_serialization:
+                    safe_save_file(
+                        output_state_dict,
+                        os.path.join(output_dir, SAFETENSORS_WEIGHTS_NAME),
+                        metadata={"format": "pt"},
+                    )
+                else:
+                    torch.save(output_state_dict, os.path.join(output_dir, WEIGHTS_NAME))
+
+                # save the config and change the inference mode to `True`
+                if peft_config.base_model_name_or_path is None:
+                    peft_config.base_model_name_or_path = (
```

Comment:
Here the `peft_config` passed by the user is mutated. Do we want that? I think it's okay if `peft_config.base_model_name_or_path` being `None` has to be a mistake, but if it's a valid value, should it be changed?

Reply:
I think this is correct as it adds the correct `base_model_name_or_path` attribute in case it is not in the config. For models that are in the HF scope (transformers & diffusers) they should have the attribute `name_or_path` if my understanding is correct but I need to double check.